fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -20,9 +20,20 @@
 reference = 23
 
 class C(object):
+    def foo(self): pass
     @staticmethod
-    def foo(): pass
-def bar(): pass
+    def foo_staticmethod(): pass
+    @classmethod
+    def foo_classmethod(cls): pass
+
+def fake(arg): pass
+def fake2(arg): pass
+def fake_no_args(): pass
+@staticmethod
```

Comment:
For instance, what should this do:
```
class F:
    @classmethod
    def foo(cls, *args):
        assert cls is F
```

when you patch F.foo onto something else? 

The effect of staticmethod is to discard the instance when calling the method; of classmethod to take the class parameter from __get__ and pass it to the method.

That implies that anything which we receive and is already a classmethod will receive the class of the object we've patched onto - unless we take the bound version ourselves - so back to the expected behaviour :).